### PR TITLE
Drone code

### PR DESCRIPTION
**Pull request by bfishbaum**
Added backend connection to drone API along with dispatching and order management.

### CODE REVIEW

File: express/droneAPIManager.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        return true
+      } else {
+        return false
+      }
+    }).catch(res => {
+      return false
+    })
+}
+
+const getDrone = (droneID, next) => {
```
Comment (machang19) [inline express/droneAPIManager.js]: I am confused as to what next is.

File: express/droneAPIManager.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          next(drone.id)
+        }
+      })
+    }
+  }
+}
+
+const getFreeDroneID = (next) => {
+  for(var i = 1; i <= maxAirbaseID; i++) {
+    getAirbase(i, (airbase) => getFreeDroneInAirbase(airbase, next))
```
Comment (machang19) [inline express/droneAPIManager.js]: I believe we should only use our airbase not loop through all possible airbases

File: express/routes/api.js [hunk 3]
```diff
@@ -151,86 +106,59 @@ apiRouter.post("/addUser", function(req, res, next) {
...
 	} else {
 		res.json(unloadedDatabaseResponse)
 	}
 })
 
+apiRouter.get("/getFreeDrone", function(req, res, next) {
+	droneManager.getFreeDrone((id) => res.json(successResponse({droneID: id})))
+})
+
+apiRouter.get("fillOrder", function(req, res, next) {
```
Comment (machang19) [inline express/routes/api.js]: This should be post request

Comment (machang19) [reply express/routes/api.js]: Also should it be /fillOrder

File: express/droneAPIManager.js (hunk 1 above)
Comment (bfishbaum) [reply express/droneAPIManager.js]: next is the callback, I'll change the variable name to callback

File: express/droneAPIManager.js (hunk 2 above)
Comment (bfishbaum) [reply express/droneAPIManager.js]: Done

File: express/routes/api.js (hunk 3 above)
Comment (bfishbaum) [reply express/routes/api.js]: nice catch!

**Comment by machang19** [comment]
+1

**Pull request merged**